### DISCUSSION TITLE
Mirroring a Windows production on a Mac

### DISCUSSION THREAD

**Discussion by kcits** [BretFisher/ama#139] [Q&A]
Our production is ColdFusion on Windows Server, running on IIS with MSSQL DB servers.  This used to not be a problem when I was using a Windows PC.

Now that I am on a Mac I am looking for a solution to mirror or production environment.

Can I use Docker to be my Dev of a Windows machine, running IIS and ColdFusion, with a MSSQL DB Server too?  And be able to access the website files via Atom locally on my Mac (no FTP or
moving of files from local to server and vise versa)?

I have tried several times and I may get one piece of this puzzle working, but then I run into either connection issues, or I am unable to access the files...

So, first, is it possible to use Docker to mirror our production environment?  If not, Thank you for your time :) If it can, can you point me in the right direction of how to utilize Docker to mirror our production environment?

**Comment by BretFisher** [comment]
Hey there, I'm not sure what your goal is... to develop in an IDE on mac and have coldfusion/iis and mssql run in windows containers somehow?

One of your big issues is "nested virtualization". You're on a mac wanting to run a windows VM with Docker Desktop inside it which will then want to run docker in a Hyper-V of Windows Server. That "virtualization inside virtualization" is known as nested virtualization. It's hard to get working and often breaks, especially with Docker.

Basically, you're choosing to have a very complex setup that may not work the way you want. Your easy way to develop on windows containers with docker is to move back to Windows as your main OS on real hardware. Then docker with windows containers will work as expected (file sharing, etc. all work).

However, there is hope in just the last few months. Since Windows 10 started supporting windows containers 2 years ago, Hyper-V had to run a VM inside win10 with the windows server kernel. But recently, using Docker Desktop on Windows 10 Pro/Ent with the latest Win updates, you might get it to work using "process isolation" which tells docker on windows 10 to *not* create the server VM, but rather try to run the process native on windows 10. This may not work with IIS/coldfusion or other apps like you expect, and it's bleeding edge... but check this out: 

https://stefanscherer.github.io/how-to-run-lightweight-windows-containers-on-windows-10/
